### PR TITLE
feat(auth): Remove default is_managed state from SSO accounts

### PR DESCRIPTION
**Pull request by dcramer**
This changes the behavior of automatic merging to always happen if emails are verified and removes the use of is_managed for SSO.

### CODE REVIEW

**Comment by dcramer** [comment]
The idea behind is_managed originally was "if this account is controlled by a third party, you shouldn't be able to change certain attributes [password, username, email]". That said, SSO accounts are not really intended to be that. They **might** be, but given Sentry accounts merge, that behavior makes less and less sense.

File: src/sentry/auth/provider.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def build_identity(self, state):
         >>>     "id": "foo@example.com",
         >>>     "email": "foo@example.com",
         >>>     "name": "Foo Bar",
+        >>>     "email_verified": True,
```
Comment (dcramer) [inline src/sentry/auth/provider.py]: https://github.com/getsentry/sentry-auth-google/pull/25

File: src/sentry/auth/helper.py [hunk 2]
```diff
@@ -492,7 +491,7 @@ def _handle_unknown_identity(self, identity):
             # the email matches we go ahead and let them merge it. This is the
             # only way to prevent them having duplicate accounts, and because
             # we trust identity providers, its considered safe.
-            if existing_user and existing_user.is_managed and verified_email:
```
Comment (dcramer) [inline src/sentry/auth/helper.py]: so the behavior is changed a bit here, and assuming tests dont fail immediately I might need to update them (forgot about that before pushing)

**Comment by dcramer** [comment]
@EvanPurkhiser i think this is in a good spot now and I also simplified the code so hopefully its a bit more consumable

**Review by mattrobenolt — APPROVED** [review]

**Review by evanpurkhiser — APPROVED** [review]
I like the bit of simplification, looks good

**Pull request merged**